### PR TITLE
feat: support custom npm clients (`options.npm`)

### PR DESCRIPTION
**Pull request by lcxfs1991**
some company has their own npm command with a different name ,like tnpm ,cnpm and so on.

### CODE REVIEW

**Comment by lcxfs1991** [comment]
Seems that the project is inactive. Wondering when all the pull request will be merged and the new ver will be published. We are looking forward to it.

**Comment by lcxfs1991** [comment]
also add webpack 3 support in this pr.

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "npm-install-webpack-plugin",
-  "version": "4.0.4",
+  "version": "4.0.5",
```
Comment (michael-ciniawsky) [inline package.json]: Please revert this, it is the maintainer/releaser responsibility to do and we have a standard process for this in place already :)

**Comment by michael-ciniawsky** [comment]
@ericclemmons @insin Any objections against this change ?

**Comment by lcxfs1991** [comment]
@michael-ciniawsky just revert the version number please have a look.

**Comment by michael-ciniawsky** [comment]
@lcxfs1991 Thx

**Pull request merged**